src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -285,7 +300,133 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+    
+    @POST
+    @Path("{id}/metadata")
+    public Response updateFileMetadata(@FormDataParam("jsonData") String jsonData,
+                    @PathParam("id") String fileIdOrPersistentId
+        ) throws DataFileTagException, CommandException { //MAD: should catch these?
+        
+        try {
+            DataverseRequest req = createDataverseRequest(findUserOrDie());
+            final DataFile df = execCommand(new GetDataFileCommand(req, findDataFileOrDie(fileIdOrPersistentId)));
+
+            //Much of this code is taken from the replace command, simplified as we aren't actually switching files
+            User authUser;
+            try {
+                authUser = findUserOrDie();
+            } catch (AbstractApiBean.WrappedResponse ex) {
+                return error(Response.Status.FORBIDDEN, 
+                        BundleUtil.getStringFromBundle("file.addreplace.error.auth")
+                        );
+            }
 
+            //You shouldn't be trying to edit a datafile that has been replaced
+            //We get the data file with a previousDataFileId of the user passed to update
+            List<Long> result = em.createNamedQuery("DataFile.findDataFileThatReplacedId", Long.class)
+            .setParameter("identifier", df.getId())
+                    .getResultList();
+            //There will be either 0 or 1 returned dataFile Id. If there is 1 this file is replaced and we need to error.
+            if(null != result && result.size() > 0) {
+                //we get the data file to do a permissions check, if this fails it'll go to the WrappedResponse below
+                execCommand(new GetDataFileCommand(req, findDataFileOrDie(result.get(0).toString())));
+                //If you are permitted we return a nicer message about the metadata error
+                return error(Response.Status.BAD_REQUEST, "You cannot edit metadata on a dataFile that has been replaced. Please try again with the newest file id.");
+            }
```

Comment:
Not sure why you're preventing an update here. If I replace a file I can update its metadata via the interface on the files page.

Reply:
Raman had prevented repeated replaces via API and I ran into an error when I tried to allow an update after replace via API. I don't remember what the error is. I decided not to fix it but I can take another look at it.

Reply:
I deleted my previous message, that code is actually in there because we don't accept updates of files that don't exist in the current version, e.g. files that have been replaced. The new file after the replace works fine.

I'm updating the documentation and maybe the error message around it.